src/passes/Print.cpp

```diff
@@ -96,9 +96,27 @@ struct TypeNamePrinter {
   void print(const Struct& struct_);
   void print(const Array& array);
   void print(const Rtt& rtt);
+
+  // FIXME: This limit avoids infinite recursion and ensures reasonable-sized
+  // output, but we should have a better mechanism for this.
```

Comment:
We already have a mechanism for avoiding infinite recursions. If you want to avoid introducting a new `prints` variable here, you could reuse the existing `currHeapTypeDepth`, but perhaps what you have is better because it also avoids printing very wide structs fully.

```suggestion
  // FIXME: This limit ensures reasonable-sized
  // output, but we should have a better mechanism for this.
```

Reply:
Ok, it is indeed not infinitely recursing... just 1,336,335 times (!)

Reply:
🤯